entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md

```diff
@@ -356,6 +505,58 @@ This change was made to avoid erroneously defining a primary key property on the
 
 If the property was intended to be the foreign key, and hence part of the primary key, then explicitly configure it as such.
 
+## Database connection is now closed if not used anymore before the TransactionScope has been completed
+
+[Tracking Issue #14218](https://github.com/aspnet/EntityFrameworkCore/issues/14218)
+
+This change will be introduced in EF Core 3.0-preview 4.
+
+**Old behavior**
+
+Before EF Core 3.0, if the context was created with a closed connection it will be maintained opened while the current `TransactionScope` is active.
+
+```C#
+using (new TransactionScope())
+{
+    using (AdventureWorks context = new AdventureWorks())
+    {
+        context.ProductCategories.Add(new ProductCategory());
+        context.SaveChanges();
+
+        // Old behavior: Connection is still open at this point
+		
+        var categories = context.ProductCategories().ToList();
+    }
+}
+```
+
+**New behavior**
+
+Starting with 3.0, EF Core will close the connection as soon as it's done using it.
+
+**Why**
+
+This change was made to make the behavior more predictable and match EF6.
+
+**Mitigations**
+
+If the connection needs to remain open explicit call to `OpenConnection()` will ensure that EF Core doesn't close it prematurely:
+
+```C#
+using (new TransactionScope())
+{
+    using (AdventureWorks context = new AdventureWorks())
+    {
+        context.Database.OpenConnection();
+        context.ProductCategories.Add(new ProductCategory());
+        context.SaveChanges();
+		
+        var categories = context.ProductCategories().ToList();
+        context.Database.CloseConnection();
```

Comment:
Should we add try/finally here to close the connection safely? On the fence.

Reply:
If an unhandled exception is thrown it doesn't matter whether the connection is closed explicitly or not. If the exception is handled (e.g. by a retry) then the connection state should be explicitly set anyway.